specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/BillingMeters.json

```diff
@@ -0,0 +1,126 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2022-06-01-preview",
+    "title": "ContainerApps API Client"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.App/locations/{location}/billingMeters": {
+      "get": {
+        "tags": [
+          "BillingMeters"
+        ],
+        "summary": "Get billing meters by location.",
+        "description": "Get all billingMeters for a location.",
+        "operationId": "BillingMeters_Get",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/BillingMeterCollection"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List billing meters by location": {
+            "$ref": "./examples/BillingMeters_Get.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "BillingMeter": {
```

Comment:
>BillingMeter [](http://example.com/codeflow?start=5&length=12)

This needs to be defined as a Proxyresource with all the proxy resource properties

Reply:
The problem with that is we need to return a list of billing meters. 
With the top level Proxyresource **properties** only being able to represent a single object (not a collection of them), I am not sure how to achieve that way.

Reply:
You are returning BillingMeterCollection as the response , it should be a value with array of BillingMeter along with a nextlink pointing to the next page as applicable ... See here for details : 

[https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#pagination](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#pagination)

Reply:
each billing meter should be a proxy resource in itself which was my original comment .

Reply:
@raosuhas - we did not use the proxy resource type envelope for the response because we did not want this to look like a full CRUD API for BillingMeter resource type. We wanted this to be just one API that always returns a collection - there is no API to get individual billing meter for example.

We are ok changing the schema to use the proper proxy resource type envelope but wanted to check if it will be ok from ARM perspective to have a GET collection API only for a resource type. There will be no create/update/delete/read single element APIs.

Reply:
To add to Ruslan's comment: because we do not have an API to get an individual billing meter what should the "id" field (that's supposed to be a unique URL for an individual resource/entity) be?

Reply:
If your scenario doesn't allow for full CRUD and a proxy resource envelope on billingMeters, you should model this as a POST action on a real resource type, e.g.
```
/subscriptions/{subscriptionId}/providers/Microsoft.App/locations/{location}/listBillingMeters
```

Response to POST can be whatever you like within MS REST guidelines.

Reply:
This endpoint is a GET on a resource collection. What we are trying to do here is list a configuration of the deployment for a specific region. There wont be any changes to this configuration after the deployment and if the configuration needs to change, that would mean we will make a change in our deployment configuration and redeploy.
So the most applicable MS REST guideline would be:
✔️ YOU MAY expose an operation that lists your resources by supporting a GET method with a URL to a resource-collection (as opposed to a resource-id).

We are okay with fitting this to location based proxy resource model.

Reply:
Yes, good choice.